Test/Campus.Master.IntegrationTests/ProfileController/CreateProfileTests.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+using Campus.Infrastructure.Data.EntityFrameworkCore.Context;
+using Campus.Master.API.Helpers.Contracts;
+using Campus.Master.IntegrationTests.Utils;
+using Campus.Services.Interfaces.DTO.Profile;
+using Campus.Services.Interfaces.Interfaces;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+using Controller =  Campus.Master.API.Controllers.ProfileController;
+
+namespace Campus.Master.IntegrationTests.ProfileController
+{
+    public class CreateProfileTests : IDisposable
+    {
+        private Controller Sut { get; }
+        private CampusContext Context { get; }
+        
+        public CreateProfileTests()
+        {
+            var collection = ServiceCollectionBuilder.BuildCollection();
+            
+            collection.AddDbContext<CampusContext>(
+                options => options.UseInMemoryDatabase(databaseName: nameof(CreateProfileTests)));
+            
+            var provider = collection.BuildServiceProvider();
+            
+            var accessor = ContextAccessorBuilder.Build("1", "1");
+
+            var profile = ServiceLocator.Get<IProfileService>(provider);
+            var token = ServiceLocator.Get<ITokenBuilder>(provider);
+            var context = ServiceLocator.Get<CampusContext>(provider);
+
+            context.AddSampleUsers();
+            Context = context;
+            Sut = new Controller(accessor, profile, token);
+        }
+        
+        [Fact]
+        public async Task ShouldThrowException_WhenPasswordsDoNotMatch()
+        {
+            // Arrange
+            var profile = new ProfileRegistrationDto
+            {
+                Email = "example@domain.com",
+                Password = "password",
+                ConfirmPassword = "password1",
+                FirstName = "FirstName",
+                LastName = "LastName"
+            };
+
+            // Act
+            Func<Task> sutCall = async () => await Sut.CreateProfile(profile);
+
+            // Assert
+            await sutCall.Should().ThrowAsync<ApplicationException>()
+                .WithMessage("Wrong username or password.");
```

Comment:
We need to consider replacement of exception message "Wrong username or password" to "Wrong email or password"